File: qubgrademe-totalmarks_marksfornext/src/validate.js
const validate = (modules, marks) => {
  let success = true;
  let message = "";

  // Checking if all modules are empty
  const areModulesEmpty = modules.reduce(
    (prev, curr) => (prev ? (curr ? curr.trim().length === 0 : false) : prev),
    true,
  );
  if (areModulesEmpty) {
    success = false;
    message = "Please provide some module codes and their respective marks";
  }

  if (success) {
    for (let i = 0; i < 5; i++) {
      // Checking if there is a module at this index
      if (modules[i] && modules[i].trim().length > 0) {
        // If there is checking that there is a valid, corresponding integer for the mark
        if (!isStringPositiveInteger(marks[i])) {
          message = "Please provide a valid integer for every entered module";
          success = false;
          break;
        }
      }
      // If there isnt a module present at this index
      // but there is a valid mark
      if (isStringPositiveInteger(marks[i])) {
        message = "Please provide a module name for all marks entered";
        success = false;
      }
    }
  }

  return { success, message };
};

const isStringPositiveInteger = (string) => {
  const num = parseInt(string);
  if (!Number.isNaN(num)) {
    return num >= 0 && num <= 100;
  }
};

module.exports = validate;

File: qubgrademe-totalmarks/src/validate.go
package main

import (
	"strconv"
	"strings"
)

func validate(modules []string, marks []string) string {

	// Check all modules are empty
	if allModulesEmpty(modules) {
		return "Please provide some module codes and their respective marks"
	}

	// Foreach module, check if there is a valid mark
	for i := 0; i < len(modules); i++ {
		if strings.TrimSpace(modules[i]) != "" {
			// Check if mark at location is valid
			if !validMark(marks[i]) {
				return "Please provide a valid integer for every entered module"
			}
		} else {
			// If not a valid module, ensure there isnt a valid mark here
			if validMark(marks[i]) {
				return "Please provide a module name for all marks entered"
			}
		}
	}
	return ""
}

func allModulesEmpty(modules []string) bool {
	for i := 0; i < len(modules); i++ {
		if strings.TrimSpace(modules[i]) != "" {
			return false
		}
	}
	return true
}

func validMark(mark string) bool {

	if mark, err := strconv.ParseInt(mark, 10, 32); err == nil {
		if mark <= 100 && mark >= 0 {
			return true
		}
	}
	return false
}
